devito/passes/equations/linearity.py

```diff
@@ -52,7 +52,9 @@ def _(c, deriv):
 
 @_is_const_coeff.register(sympy.Function)
 def _(c, deriv):
-    return not set(deriv.dims) & c.free_symbols
+    c_dims = set().union(*[getattr(i, '_defines', i) for i in c.free_symbols])
```

Comment:
`isinstance(i, Dimension)` makes it more clear. `getattr` should be avoided whenever possible as it hides information away

Reply:
also, it'd seem that if you don't change it to use `isinstance` you would end up adding other non-Dimension objects to `c_dims`?

Reply:
I don't think these extras are problematic, it won't e a match anyway. I just prefer `getattr` to local import that's all but can change.

Reply:
hang on, why would it be a local import?

Reply:
There is a circular dependency somewhere (didn't go to the bottom of it just switched back to isinstance) so can't import Dimension here unless inside the function.